modules/calib3d/test/test_usac.cpp

```diff
@@ -0,0 +1,326 @@
+        // use two relative scenes. The first camera is P1 = K1 [I | 0], the second P2 = K2 [R | t]
+        pts1 = K1 * points3d;
+        pts2 = K2 * (R * points3d + t * cv::Mat::ones(1, points3d.cols, points3d.type()));
+
+        // normalize by 3 coordinate
+        cv::divide(pts1.row(0), pts1.row(2), pts1.row(0));
+        cv::divide(pts1.row(1), pts1.row(2), pts1.row(1));
+        cv::divide(pts2.row(0), pts2.row(2), pts2.row(0));
+        cv::divide(pts2.row(1), pts2.row(2), pts2.row(1));
+
+        // get 2D points
+        pts1 = pts1.rowRange(0,2); pts2 = pts2.rowRange(0,2);
+
+        // generate random outliers as 2D image points
+        cv::Mat pts1_outliers(pts1.rows, out_size, pts1.type()),
+                pts2_outliers(pts2.rows, out_size, pts2.type());
+        rng.fill(pts1_outliers, cv::RNG::UNIFORM, 0, 1000);
+        rng.fill(pts2_outliers, cv::RNG::UNIFORM, 0, 1000);
+        // for epipolar geometry merge inliers and outliers
+        cv::hconcat(pts1, pts1_outliers, pts1);
+        cv::hconcat(pts2, pts2_outliers, pts2);
+
+        // add normal / Gaussian noise to image points
+        cv::Mat noise1 (pts1.rows, pts1.cols, pts1.type()), noise2 (pts2.rows, pts2.cols, pts2.type());
+        rng.fill(noise1, cv::RNG::NORMAL, 0, noise_std); pts1 += noise1;
+        rng.fill(noise2, cv::RNG::NORMAL, 0, noise_std); pts2 += noise2;
+    }
+
+    return inl_size;
+}
+
+double getError (TestSolver test_case, int pt_idx, const cv::Mat &pts1, const cv::Mat &pts2, const cv::Mat &model) {
+    cv::Mat pt1 = pts1.col(pt_idx), pt2 = pts2.col(pt_idx);
+    if (test_case == TestSolver::Homogr) { // reprojection error
+        // compute Euclidean distance between given and reprojected points
+        cv::Mat est_pt2 = model * pt1; est_pt2 /= est_pt2.at<double>(2);
+        if (false) {
+            cv::Mat est_pt1 = model.inv() * pt2; est_pt1 /= est_pt1.at<double>(2);
+            return (cv::norm(est_pt1 - pt1) + cv::norm(est_pt2 - pt2)) / 2;
+        }
+        return cv::norm(est_pt2 - pt2);
+    } else
+    if (test_case == TestSolver::Fundam || test_case == TestSolver::Essen) {
+        cv::Mat l2 = model     * pt1;
+        cv::Mat l1 = model.t() * pt2;
+        if (test_case == TestSolver::Fundam) // sampson error
+            return pow(pt2.dot(l2),2) / (pow(l1.at<double>(0), 2) + pow(l1.at<double>(1), 2) +
+                                         pow(l2.at<double>(0), 2) + pow(l2.at<double>(1), 2));
+        else // symmetric geometric distance
+            return (fabs(pt1.dot(l1)) / sqrt(pow(l1.at<double>(0),2) + pow(l1.at<double>(1),2)) +
+                    fabs(pt2.dot(l2)) / sqrt(pow(l2.at<double>(0),2) + pow(l2.at<double>(1),2)))/2;
+    } else
+    if (test_case == TestSolver::PnP) { // PnP, reprojection error
+        cv::Mat img_pt = model * pt2; img_pt /= img_pt.at<double>(2);
+        return cv::norm(pt1 - img_pt);
+    } else
+        CV_Error(1, "undefined test_case");
```

Comment:
> 1

What is 1?

Prefer to use GoogleTest for errors reporting in test code:
```
FAIL() << msg;
```

Reply:
I meant 1 to be something like EXIT_FAILURE. I will use GoogleTest error reporting then.